src/stacked_borrows.rs

```diff
@@ -764,8 +797,16 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                         Permission::SharedReadWrite
                     };
                     let item = Item { perm, tag: new_tag, protector };
-                    stacked_borrows.for_each(range, |offset, stack| {
-                        stack.grant(orig_tag, item, (alloc_id, range, offset), &*global)
+                    let mut global = this.machine.stacked_borrows.as_ref().unwrap().borrow_mut();
```

Comment:
This moves a `borrow` into the closure, which is probably bad for performance.
Is it possible to keep it at the old place, or does that lead to multiple borrows?

Reply:
Yeah it does lead to multiple borrows. I actually attempted this transformation before in a previous commit then I amended it out when CI failed.

Reply:
Okay, then please add a comment explaining that.